parsec/core/cli/share_workspace.py

```diff
@@ -13,24 +13,36 @@
 WORKSPACE_ROLE_CHOICES = {"NONE": None, **{role.value: role for role in WorkspaceRole}}
 
 
-async def _share_workspace(config, device, name, user_id, user_role):
+async def _share_workspace(config, device, name, user_id, recipiant, user_role):
+    if recipiant and user_id or not recipiant and not user_id:
+        raise click.ClickException("Either recipiant or user_id should be used, but not both")
     async with logged_core_factory(config, device) as core:
         workspace = core.find_workspace_from_name(name)
+        if recipiant:
+            user_info_tab, nb = await core.find_humans(recipiant, 1, 100, False, False)
+            if nb == 0:
+                raise RuntimeError("This recipiant doesn't exist, use the email address")
```

Comment:
I think "Unknown recipiant" is good enough (the "use the email address" part is misleading if the command has already been used with an email address that happened to not be part of the existing human handles)